app/models/asset_field_type.rb

```diff
@@ -74,6 +75,16 @@ def allowed_content_types
     end
   end
 
+  def asset_extension
+    unless existing_data['asset']['content_type'] == asset_content_type
+     errors.add(:asset, "Asset must be the same type of: #{existing_data['asset']['content_type']}")
+    end
+  end
+
+  def asset_field_type_id
```

Comment:
I'd suggest merging in the Media Title story, then rebasing this branch with `develop`. Also, would you mind updating the PR title to follow guidelines?